src/amm/amm.controllers.ts

```diff
@@ -75,20 +80,23 @@ import {
 import { Algorand } from '../chains/algorand/algorand';
 import { Tinyman } from '../connectors/tinyman/tinyman';
 import { Plenty } from '../connectors/plenty/plenty';
+import { QuipuSwap } from '../connectors/quipuswap/quipuswap';
 
 export async function price(req: PriceRequest): Promise<PriceResponse> {
   const chain = await getInitializedChain<
     Algorand | Ethereumish | Nearish | Tezosish
   >(req.chain, req.network);
-  const connector: Uniswapish | RefAMMish | Tinyman | Plenty =
-    await getConnector<Uniswapish | RefAMMish | Tinyman | Plenty>(
+  const connector: Uniswapish | RefAMMish | Tinyman | Plenty | QuipuSwap =
```

Comment:
Does QuipuSwap have be a new controller type? We want to avoid this since it bloats the codebase.

I understand that new chains have idiosyncrasies that require customization, but can I think this connector should be able to use the Plenty controller spec, since it's also a Tezos AMM

Reply:
Hey @fengtality, yeah both of them work on Tezos. But unfortunately, they both are slightly different on-chain, and very different off-chain. I wished one of them to be a fork of the other, but that isn't the case here. The biggest differences that I found are in their API, which I can't change.
We can try to abstract some logic from them, but for only two connectors it may bloat the code even more.